awsagentprovider/src/test/java/software/amazon/opentelemetry/javaagent/providers/AwsMetricAttributeGeneratorTest.java

```diff
@@ -757,6 +761,32 @@ public void testSdkClientSpanWithRemoteResourceAttributes() {
     mockAttribute(GEN_AI_REQUEST_MODEL, "test.service_^id");
     validateRemoteResourceAttributes("AWS::Bedrock::Model", "test.service_^^id");
     mockAttribute(GEN_AI_REQUEST_MODEL, null);
+
+    // Validate behaviour of AWS_STATE_MACHINE_ARN attribute, then remove it.
+    mockAttribute(
+        AWS_STATE_MACHINE_ARN,
+        "arn:aws:states:us-east-1:123456789012:stateMachine:test_state_machine");
+    validateRemoteResourceAttributes("AWS::StepFunctions::StateMachine", "test_state_machine");
+    mockAttribute(AWS_STATE_MACHINE_ARN, null);
+
+    // Validate behaviour of AWS_STEPFUNCTIONS_ACTIVITY_ARN, then remove it.
+    mockAttribute(
+        AWS_STEP_FUNCTIONS_ACTIVITY_ARN,
+        "arn:aws:states:us-east-1:007003123456789012:activity:testActivity");
+    validateRemoteResourceAttributes("AWS::StepFunctions::Activity", "testActivity");
+    mockAttribute(AWS_STEP_FUNCTIONS_ACTIVITY_ARN, null);
+
+    // Validate behaviour of AWS_SNS_TOPIC_ARN, then remove it.
+    mockAttribute(AWS_SNS_TOPIC_ARN, "arn:aws:sns:us-west-2:012345678901:testTopic");
+    validateRemoteResourceAttributes("AWS::SNS::Topic", "testTopic");
+    mockAttribute(AWS_SNS_TOPIC_ARN, null);
+
+    // Validate behaviour of AWS_SECRET_ARN, then remove it.
+    mockAttribute(
+        AWS_SECRET_ARN, "arn:aws:secretsmanager:us-east-1:123456789012:secret:secretName");
+    validateRemoteResourceAttributes("AWS::SecretsManager::Secret", "secretName");
+    mockAttribute(AWS_SECRET_ARN, null);
+
```

Comment:
TODO: Need to add unit test for Lambda.

I'm not very familiar with mocking in Java and was running into issues implementing this. I think this can be done later since I have already verified the span attributes with end-to-end testing.

Reply:
This unit test will no longer be needed since I am moving implementation to upstream (see above thread).